### PR TITLE
ESXi: add support for uefi pxe install and ipxe

### PR DESCRIPTION
**Pull request by javier-angulo**
This PR tries to add support for ESXi UEFI installs, as well as iPXE. We currently are running cobbler 3.2 with some ugly hacks to install ESXi with UEFI firmware, and we are very interested to have this feature upstream. 

Changes:
- create new 'esxi' folder on tftpboot to hold esxi bootloaders, boot.cfg files, links to images and pxelinux.cfg.
- distro_signatures: for esxi distros, add supported boot_loaders, removed template_files
- moved all esxi kernel options to boot.cfg
- for UEFI install, create dir on tftp with boot.cfg and symlink to esxi UEFI bootloader for each mac address
- added system-test for esxi import
- documented esxi install in user-guide

What works (dhcpv4):
- pxe bios install (via syslinux-3.86 pxelinux.0 and mboot.c32)
- pxe uefi install (via ESXi UEFI bootloader mboot.efi )
- ipxe bios install (via undionly.kpxe chainloading syslinux-3.86 pxelinux.0)
- ipxe uefi install (via snponly.efi chainloading ESXi UEFI bootloader mboot.efi)

What does not work:
- use DHCPv6 to install ESXi.
- UEFI firmware HTTP install
- Boot menu for profiles. Can be made to work for BIOS PXE, though requires changes on dhcp.template

### CODE REVIEW

File: system-tests/tests/svc-bootcfg [hunk 1]
```diff
@@ -5,37 +5,46 @@ source ${SYSTESTS_PRELUDE} && prepare
 
 set -x -e -o pipefail
 
-# Arrange
-cobbler distro add --name fake --arch x86_64 --kernel ${fake_kernel} \
-	--initrd ${fake_initramfs}
-cobbler profile add --name fake --distro fake
-cobbler system add --name testbed --profile fake
+distro=esxi70
```
Comment (SchoolGuy) [inline system-tests/tests/svc-bootcfg:8]: Can we maybe add a second test for this case? I would desire to test both cases. The "standard one" I had before and the ESXi case you added here. I would propose just copy the file you have now and suffix it with `-esxi`, then you revert the changes in this file here.

File: cobbler/tftpgen.py [hunk 2]
```diff
@@ -1517,3 +1535,51 @@ def _generate_initrd(
...
+                try:
+                    os.symlink("../../mboot.efi", temp_link_file)
+                    break
+                except FileExistsError:
+                    pass
+            try:
+                os.replace(temp_link_file, link_file)
+            except OSError:
+                os.remove(temp_link_file)
+                raise OSError("Error creating symlink %s" % link_file)
```
Comment (SchoolGuy) [inline cobbler/tftpgen.py:1583]:
```suggestion
            except OSError as os_error:
                os.remove(temp_link_file)
                raise OSError("Error creating symlink %s" % link_file) from os_error
```

**Comment by SchoolGuy** [comment]
@javier-angulo I need to check some thing I am not sure of but I will get back to this PR this week. It is a huge contribution and I am very thankful for it! If your company is interested I would feature it [here](https://cobbler.github.io/users.html).

**Comment by javier-angulo** [comment]
@SchoolGuy thank you for comments, much appreciated. I am a sysops guy, so please review my code carefully.

File: system-tests/tests/svc-bootcfg (hunk 1 above)
Comment (javier-angulo) [reply system-tests/tests/svc-bootcfg:8]: I will do that, but I will not be able keep the exact previous test without some minor modifications, as `bootcfg.template` has changed.

File: cobbler/tftpgen.py (hunk 2 above)
Comment (javier-angulo) [reply cobbler/tftpgen.py:1583]: Thank you, will do.

**Comment by javier-angulo** [comment]
@SchoolGuy did you have a chance to look at this PR? Just let me know if you need further clarifications.

**Comment by SchoolGuy** [comment]
@javier-angulo So code wise I believe we are good. I don't understand everything but enough to say that it can be merged. Before we do, can you maybe have a look at the following Issues/PRs?

- https://github.com/cobbler/cobbler/issues/2337
- https://github.com/cobbler/cobbler/issues/2889
- https://github.com/cobbler/cobbler/pull/2897

Those are related to ESXi and I would love if we can either close or improve them in a way that I know what is left to do there.

**Comment by javier-angulo** [comment]
#2337 is definitively addressed, apart from UEFI httpboot. I am planning to have a look on this later on, but currently only grub and images tftp folders are exported via http, and I would need to export the esxi folder as well.
#2889 was already reported and solved on #2480. I have not had this issue since (needs python-magic). The OP is using fedora 35, maybe a problem with the packaging? On centos8 works out of the box with python3-magic.
#2897 I believe is not needed per #2480

**Review by SchoolGuy — APPROVED** [review]
LGTM. Thanks for the big contribution!
